packages/gatsby/src/joi-schemas/__tests__/joi.ts

```diff
@@ -140,6 +140,90 @@ describe(`gatsby config`, () => {
       `[ValidationError: "trailingSlash" must be one of [always, never, ignore, legacy]]`
     )
   })
+
+  it(`return false for graphqlTypegen when not set`, () => {
+    const config = {}
+
+    const result = gatsbyConfigSchema.validate(config)
+    expect(result.value).toEqual(
+      expect.objectContaining({
+        graphqlTypegen: false,
+      })
+    )
+  })
+
+  it(`throws when graphqlTypegen is not valid option`, () => {
+    const config = {
+      graphqlTypegen: `foo-bar`,
+    }
+
+    const result = gatsbyConfigSchema.validate(config)
+    expect(result.error).toMatchInlineSnapshot(
+      `[ValidationError: "graphqlTypegen" must be one of [boolean, object]]`
+    )
+  })
+
+  it(`throws when graphqlTypegen has invalid keys`, () => {
+    const config = {
+      graphqlTypegen: {
+        invalid: true,
+      },
+    }
+
+    const result = gatsbyConfigSchema.validate(config)
+    expect(result.error).toMatchInlineSnapshot(
+      `[ValidationError: "graphqlTypegen.invalid" is not allowed]`
+    )
+  })
+
+  it(`return defaults for graphqlTypegen when empty object is set`, () => {
+    const config = {
+      graphqlTypegen: {},
+    }
+
+    const result = gatsbyConfigSchema.validate(config)
+    expect(result.value).toEqual(
+      expect.objectContaining({
+        graphqlTypegen: {
+          typesOutputPath: `src/gatsby-types.d.ts`,
+        },
+      })
+    )
+  })
+
+  it(`return defaults for graphqlTypegen when true is set`, () => {
+    const config = {
+      graphqlTypegen: true,
+    }
+
+    const result = gatsbyConfigSchema.validate(config)
+    expect(result.value).toEqual(
+      expect.objectContaining({
+        graphqlTypegen: {
+          typesOutputPath: `src/gatsby-types.d.ts`,
+        },
+      })
+    )
+  })
+
+  // Re-Add when more options are added
+  it.skip(`return partial defaults for graphqlTypegen when partial options object is set`, () => {
```

Comment:
Should this be in this PR? Not big deal if there is a follow-up already in-progress / planned, just odd to add skipped test for "problems" not yet existing

Reply:
No strong feelings about it, I just wanted to add before I forget it. There's currently not more planned

Reply:
Made it a TODO comment instead